EKF/ekf_helper.cpp

```diff
@@ -910,7 +910,11 @@ void Ekf::get_innovation_test_status(uint16_t &status, float &mag, float &vel, f
 	pos = math::max(sqrtf(_gps_pos_test_ratio(0)), sqrtf(_ev_pos_test_ratio(0)));
 
 	// return the vertical position innovation test ratio
-	hgt = sqrtf(_gps_pos_test_ratio(0));
+	hgt = sqrtf(math::max(math::max(_gps_pos_test_ratio(1),
```

Comment:
Shouldn't we be returning the corresponding test ratio for the current height mode?

Reply:
To be sure this return the current height ratio, I added a reset of all height ratios before calculating the current one .

I can use `ratio._control_status.flags.*_hgt` here to find the correct height ratio if you think that is better but I thought that is easier to read like that and more robust to future changes like that.

Reply:
I have a minor preference for being lazier (not blanket resetting everything every iteration regardless of use), but I think it's fine either way here.

Reply:
here is a new implementation without useless resets